pkg/flare/archive.go

```diff
@@ -302,6 +315,7 @@ func zipLogFiles(tempDir, hostname, logFilePath string, permsInfos permissionsIn
 
 			if permsInfos != nil {
 				permsInfos.add(src)
+				addParentPerms(src, permsInfos)
```

Comment:
`addParentPerms(src, permsInfos)` is called for every file in `logFileDir` folder.
Calling `addParentPerms(logFileDir, permsInfos)` at line 304 should be enough if you remove line https://github.com/DataDog/datadog-agent/pull/5686/files#diff-275e8edaaa2f10762a12d565c04a7f13R291.